### PR TITLE
844 link validation

### PR DESCRIPTION
**Pull request by kronken**
- Made updates to link validation model to reflect backend changes
- Also implemented shared component for calculated length field

### CODE REVIEW

**Comment by Divinare** [comment]
After "Laske" button is clicked, make it disabled? Or actually if the number in laske button is the same as in the input field, make the button disabled

File: src/factories/linkFactory.ts [hunk 1]
```diff
@@ -38,11 +35,8 @@ class LinkFactory {
             measuredLength: 0,
             municipalityCode: '',
             streetName: '',
-            streetNumber: '',
             modifiedBy: '',
-            modifiedOn: new Date(),
-            direction: '0',
-            osNumber: ''
+            modifiedOn: new Date()
```
[the remark this answers is not in the document]
Comment (kronken) [reply src/factories/linkFactory.ts]: This is how we done it in other views. Did now hide modifiedOn and modifiedAt from new link.

**Comment by kronken** [comment]
Yes, streetname and kunta can be left empty.
measured length will be implemented in: https://github.com/HSLdevcom/jore-map-ui/issues/892

**Comment by kronken** [comment]
Pr fixes done

File: src/factories/linkFactory.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Divinare) [reply src/factories/linkFactory.ts]: Ok, rest of the views will be done here: https://github.com/HSLdevcom/jore-map-ui/projects/1

**Pull request merged**